ocw/lib/azure.py

```diff
@@ -141,32 +160,45 @@ def parse_image_name(self, img_name):
         ]
         return self.parse_image_name_helper(img_name, regexes)
 
-    def cleanup_sle_images_container(self, bbsrv, container_name):
-        generator = bbsrv.list_blobs(container_name)
-        images = dict()
-        for img in generator:
+    def cleanup_sle_images_container(self, bbsrv, keep_images):
+        container_name = 'sle-images'
+        for img in bbsrv.list_blobs(container_name):
             m = self.parse_image_name(img.name)
-            if (m):
-                key = m['key']
-                if key not in images:
-                    images[key] = list()
-
-                logger.debug('[{}]Image {} is candidate for deletion with build {} '.format(
+            if m:
+                logger.debug('[{}] Blob {} is candidate for deletion with build {} '.format(
                     self.__credentials.namespace, img.name, m['build']))
-                images[key].append({
-                    'build': m['build'],
-                    'name': img.name,
-                    'last_modified': img.properties.last_modified,
-                })
-            else:
-                logger.error("[{}] Unable to parse image name '{}'".format(self.__credentials.namespace, img.name))
-
-        for key in images:
-            images[key].sort(key=lambda x: LooseVersion(x['build']), reverse=True)
 
-        for img_list in images.values():
-            for i in range(0, len(img_list)):
-                img = img_list[i]
-                if (self.needs_to_delete_image(i, img['last_modified'])):
-                    logger.info("[Azure] Delete image '{}'".format(img['name']))
-                    bbsrv.delete_blob(container_name, img['name'], snapshot=None)
+                if img.name not in keep_images:
+                    logger.info("[Azure] Delete image '{}'".format(img.name))
```

Comment:
wait you just said one line above that this blob not image , also same concern about `[Azure]` tag and not needed `format` call but I think I won't spam PR with mentioning this for every log call which you doing

Reply:
Maybe we sould rename `keep_images` into `keep_image_names`.
The Blob, Disk and Image have all the same name.